target_chains/stylus/contracts/pyth-receiver/src/structs.rs

```diff
@@ -20,5 +78,27 @@ pub struct PriceInfoStorage {
     pub ema_conf: StorageU64,
 }
 
+pub struct PriceInfo {
+    pub publish_time: U64,
+    pub expo: I32,
+    pub price: I64,
+    pub conf: U64,
+    pub ema_price: I64,
+    pub ema_conf: U64,
+}
+
+impl From<&PriceFeedMessage> for PriceInfo {
+    fn from(price_feed_message: &PriceFeedMessage) -> Self {
+        Self {
+            publish_time: U64::from(price_feed_message.publish_time),
+            expo: I32::from_be_bytes(price_feed_message.exponent.to_be_bytes()),
+            price: I64::from_be_bytes(price_feed_message.price.to_be_bytes()),
+            conf: U64::from(price_feed_message.conf),
+            ema_price: I64::from_be_bytes(price_feed_message.ema_price.to_be_bytes()),
+            ema_conf: U64::from(price_feed_message.ema_conf),
```

Comment:
no way to convert them directly? it's a bit strange that you need to convert them to bytes and back